chainer/functions/array/broadcast.py

```diff
@@ -123,9 +124,18 @@ def forward(self, xs):
             bx, _ = xp.broadcast_arrays(x, dummy)
             return bx,
 
-    def backward(self, xs, grads):
-        return _backward_one(
-            self._xp, self._in_shape, self._in_dtype, grads[0]),
+    def backward(self, indexes, grad_outputs):
+        gx, = grad_outputs
+        shape = self._in_shape
+        ndim = len(shape)
+        if gx.ndim != ndim:
+            gx = chainer.functions.sum(gx, tuple(range(gx.ndim - ndim)), False)
```

Comment:
I feel it's better to merge these two sums. What do you think?

Reply:
I thought the difference in `keepdims` can be caught by expand_dims/squeeze. For example, I think it's simpler and more efficient to first take sum with `keepdims=True` and then squeeze the preceeding dimensions of length 1 (corresponding to the `keepdims` of the first sum in the original code).